include/picongpu/particles/Synchrotron/AlgorithmSynchrotron.hpp

```diff
@@ -0,0 +1,428 @@
+
+                using ValueType_E = FieldE::ValueType;
+                using ValueType_B = FieldB::ValueType;
+                /* global memory EM-field and current density device databoxes */
+                PMACC_ALIGN(eBox, FieldE::DataBoxType);
+                PMACC_ALIGN(bBox, FieldB::DataBoxType);
+                /* shared memory EM-field device databoxes */
+                PMACC_ALIGN(cachedE, DataBox<SharedBox<ValueType_E, typename BlockArea::FullSuperCellSize, 1>>);
+                PMACC_ALIGN(cachedB, DataBox<SharedBox<ValueType_B, typename BlockArea::FullSuperCellSize, 0>>);
+
+                // F1F2DeviceBuff_ is a pointer to a databox containing F1 and F2 values. m stands for member
+                GridBuffer<float_X, 2>::DataBoxType m_F1F2DeviceBuff;
+                GridBuffer<int32_t, 1>::DataBoxType m_failedRequirementQBuff;
+
+            public:
+                /* host constructor initializing member : random number generator */
+                AlgorithmSynchrotron(
+                    const uint32_t currentStep,
+                    GridBuffer<float_X, 2>::DataBoxType F1F2DeviceBuff,
+                    GridBuffer<int32_t, 1>::DataBoxType failedRequirementQBuff)
+                    : randomGen(RNGFactory::createRandom<Distribution>())
+                {
+                    m_F1F2DeviceBuff = F1F2DeviceBuff;
+                    m_failedRequirementQBuff = failedRequirementQBuff;
+
+                    DataConnector& dc = Environment<>::get().DataConnector();
+                    /* initialize pointers on host-side E-(B-)field and current density databoxes */
+                    auto fieldE = dc.get<FieldE>(FieldE::getName());
+                    auto fieldB = dc.get<FieldB>(FieldB::getName());
+                    /* initialize device-side E-(B-)field and current density databoxes */
+                    eBox = fieldE->getDeviceDataBox();
+                    bBox = fieldB->getDeviceDataBox();
+                }
+
+                /** cache fields used by this functor
+                 *
+                 * @warning this is a collective method and calls synchronize
+                 *
+                 * @tparam T_Worker lockstep::Worker, lockstep worker
+                 *
+                 * @param worker lockstep worker
+                 * @param blockCell relative offset (in cells) to the local domain plus the guarding cells
+                 * @param workerCfg configuration of the worker
+                 */
+                template<typename T_Worker>
+                DINLINE void collectiveInit(const T_Worker& worker, const DataSpace<simDim>& blockCell)
+                {
+                    /* caching of E and B fields */
+                    cachedB = CachedBox::create<0, ValueType_B>(worker, BlockArea());
+                    cachedE = CachedBox::create<1, ValueType_E>(worker, BlockArea());
+
+                    /* instance of nvidia assignment operator */
```

Comment:
Its probably an alpaka instance that works with nvidia and amd 
I assume this is an outdated comment from the ionization code, right?